numpyro/distributions/transforms.py

```diff
@@ -450,6 +450,34 @@ def _inverse(self, y):
     def log_abs_det_jacobian(self, x, y, intermediates=None):
         return x
 
+class SoftplusTransform(Transform):
+    r"""
+    Transform from unconstrained space to positive domain via softplus :math:`y = \log(1 + \exp(x))`.
+    The inverse is computed as :math:`y = \log(\exp(x) - 1)`.
+    """
+    def __init__(self, domain=constraints.real):
+        self.domain = domain
+
+    @property
+    def codomain(self):
+        if self.domain is constraints.real:
+            return constraints.positive
+        elif isinstance(self.domain, constraints.greater_than):
+            return constraints.greater_than(self.__call__(self.domain.lower_bound))
+        elif isinstance(self.domain, constraints.interval):
+            return constraints.interval(self.__call__(self.domain.lower_bound),
+                                        self.__call__(self.domain.upper_bound))
+        else:
+            raise NotImplementedError
+
+    def __call__(self, x):
+        return softplus(x)
+
+    def inv(self, x):
+        return jnp.log(jnp.expm1(x))
```

Comment:
A [more stable version](https://github.com/pytorch/pytorch/pull/52300#discussion_r578900954) can be
```
jnp.log(-jnp.expm1(-y)) - y
```

Reply:
I will update soon.

This got me thinking, what are other differences between jax and pytorch from this transform. Looks like jax uses more naive implementation (https://jax.readthedocs.io/en/latest/_modules/jax/_src/nn/functions.html#softplus) of softplus than pytorch (that reverts to linear after x=20 and has beta parameter).

Interestingly, this less numerically stable version still works for cell2location.